dev/system-test/firestore.ts

```diff
@@ -2265,6 +2266,70 @@ describe('Aggregates', () => {
       await runQueryAndExpectCount(count5, 1);
     });
   }
+
+  it('sums multiple documents', async () => {
+    await randomCol.doc('doc1').set({foo: 2});
+    await randomCol.doc('doc2').set({foo: 4});
+    await randomCol.doc('doc3').set({foo: 6});
+    await randomCol.doc('doc4').set({notfoo: 8});
+
+    const aggregateQuery = randomCol.aggregate({
+      fooSum: AggregateField.sum('foo'),
+    });
+    const res = await aggregateQuery.get();
+
+    expect(res.data().fooSum).to.equal(12);
+  });
+
+  it('averages multiple documents', async () => {
+    await randomCol.doc('doc1').set({foo: 2});
+    await randomCol.doc('doc2').set({foo: 4});
+    await randomCol.doc('doc3').set({foo: 6});
+    await randomCol.doc('doc4').set({notfoo: 8});
+
+    const aggregateQuery = randomCol.aggregate({
+      fooAverage: AggregateField.average('foo'),
+    });
+    const res = await aggregateQuery.get();
+
+    expect(res.data().fooAverage).to.equal(4);
+  });
+
+  it('runs multiple aggregates over the same field', async () => {
+    await randomCol.doc('doc1').set({foo: 2});
+    await randomCol.doc('doc2').set({foo: 4});
+    await randomCol.doc('doc3').set({foo: 6});
+    await randomCol.doc('doc4').set({notfoo: 8});
+
+    const aggregateQuery = randomCol.aggregate({
+      fooAverage: AggregateField.average('foo'),
+      fooSum: AggregateField.sum('foo'),
+      count: AggregateField.count(),
+    });
+    const res = await aggregateQuery.get();
+
+    expect(res.data().fooAverage).to.equal(4);
+    expect(res.data().fooSum).to.equal(12);
+    expect(res.data().count).to.equal(3);
+  });
+
+  it('runs multiple aggregates over the multiple fields', async () => {
+    await randomCol.doc('doc1').set({foo: 2, bar: 10});
+    await randomCol.doc('doc2').set({foo: 4, bar: 20});
+    await randomCol.doc('doc3').set({foo: 6});
+    await randomCol.doc('doc4').set({notfoo: 8});
+
+    const aggregateQuery = randomCol.aggregate({
+      barAverage: AggregateField.average('bar'),
+      fooSum: AggregateField.sum('foo'),
+      count: AggregateField.count(),
```

Comment:
I wonder if `count` could be used to check how many documents a query would return and how many documents have a particular field.

e.g. 
```ts
const aggregateQuery = randomCol.aggregate({
  count: AggregateField.count('foo'),
});
const res = await aggregateQuery.get();
expect(res.data().count).to.equal(3);
```

If no field is given all documents would be counted.

Reply:
And I bet it may be technically more complex, but I think I'd expect that in this case here:

`fooSum` would still be 12, while `barAverage` be 15, as all documents containing the field `bar` average to 15 and all documents containing the field `foo` sum to 12.

And for count maybe I'd expect 3 here as only 3 documents have either the field `foo` or `bar`.